src/safemode_ui.h

```diff
@@ -0,0 +1,119 @@
+#ifndef SEFEMODE_UI_H
+#define SEFEMODE_UI_H
+
+#include <unordered_map>
+#include <string>
+#include <vector>
+#include <locale>
+#include <algorithm>
+#include "json.h"
+#include "enums.h"
+#include "creature.h"
+
+class safemode : public JsonSerializer, public JsonDeserializer
+{
+    private:
+        void test_pattern( const int iCurrentPage, const int iCurrentLine );
+
+        void load( const bool bCharacter );
+        bool save( const bool bCharacter );
+
+        bool bChar;
+
+        enum TAB : int {
+            GLOBAL_TAB,
+            CHARACTER_TAB,
+            MAX_TAB
+        };
+
+        class cRules
+        {
+            public:
+                std::string sRule;
+                bool bActive;
+                bool bExclude;
+                int attCreature;
+                int iProxyDist;
```

Comment:
Is this variable supposed to be proximity distance? `proxy` and `proximity` aren't highly related words, it would be better to use just `proximity` or `distance`.